daemon/protocol/client_lib.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+
+import os
+import re
+import tempfile
+import shutil
+
+from grpc.beta import implementations
+#import grpc
+import daemon_pb2
+
+
+# Copied from daemon/protocol/locate.go
+def _locate_scoot_dir():
+    scootdir = os.environ.get("SCOOTDIR")
+    if scootdir is not None:
+        return scootdir
+    homedir = os.environ.get("HOME")
+    if homedir is not None:
+        return os.path.join(homedir, ".scoot")
+    return None
+
+def _locate_socket():
+    scootdir = _locate_scoot_dir()
+    if scootdir is None:
+        return None
+    return os.path.join(scootdir, "socket")
+
+
+# Singleton variables.
```

Comment:
Prefer an implementation of the client as a class, where these vars are data members and the start/stop/run/etc functions are methods.  This will let us ditch the global/singleton pattern, and be able to use `__init__` to simplify the "if is_started" checks in all of the functions. This should make the client here super easy to use as a lib or call from separate test files, although I'm not sure what our desired lib vs reference implementation separation is. (Maybe all of the daemon_pb2 calls can be factored out into non-class functions that will eventually make up the library portion in their own file).  I think this would be more flexible either way, WDYT?